### PR TITLE
feat(figure): set figure to size of content and add `data-size="full"` attribute to stretch content to full width

### PR DESCRIPTION
**Pull request by MichaelSolati**
Fixes #SOME_ISSUE_NUMBER

Changes proposed in this pull request:

- set figure to size of content
- add `data-size="full"` attribute to stretch content to full width
-

When you're ready to submit your PR, don't forget to add the `$-presubmit` label.

Here is the new figure:
image

Here it is with the full size attribute:
image

### CODE REVIEW

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 1]
```diff
@@ -309,7 +309,7 @@ workbox.routing.registerRoute(navigationRoute);
 
 ## Demo
 
-<figure>
+<figure data-size="full">
```
Comment (devnook) [inline src/site/content/en/blog/mainline-mensware/index.md:312]: This is causing the border to be too big for the figure:

File: src/scss/blocks/_figure.scss [hunk 2]
```diff
@@ -2,34 +2,35 @@
 /// https://web.dev/design-system/component/figure
 figure {
   border: 1px solid var(--color-stroke);
-  padding: 1em;
   border-radius: $global-radius-base;
+  display: inline-block;
+  padding: 1em;
+
```
Comment (devnook) [inline src/scss/blocks/_figure.scss:8]: figure needs to be centered in the prose. right now it is inlined to the left sode.

**Comment by devnook** [comment]
Reviewed

File: src/scss/blocks/_figure.scss (hunk 2 above)
Comment (MichaelSolati) [reply src/scss/blocks/_figure.scss:8]: I'd be totally fine with that. I don't think the border is the most important thing. It isn't really used consistently for me to think it's a good design choice either... =/

File: src/site/content/en/blog/mainline-mensware/index.md (hunk 1 above)
Comment (MichaelSolati) [reply src/site/content/en/blog/mainline-mensware/index.md:312]: Without stretching it it actually looks worse... The video itself is fairly small, and by increasing the size of the video player to get the video to look good we end up getting a bunch of added white space.

image

**Comment by MichaelSolati** [comment]
@devnook I believe this is ready for re-review

File: src/scss/blocks/_figure.scss [hunk 3]
```diff
@@ -1,35 +1,38 @@
 /// COMPONENT LIBRARY LOCATION
 /// https://web.dev/design-system/component/figure
 figure {
-  border: 1px solid var(--color-stroke);
-  padding: 1em;
-  border-radius: $global-radius-base;
+  display: block;
+  width: 100%;
```
Comment (devnook) [inline src/scss/blocks/_figure.scss:5]: The img and caption align in "pure" figure is still not great:
Screenshot 2022-01-18 at 13 35 04

How about you add "text-align: center;" to the non-sized and non-floated figurecaption? The outcome should be:

Screenshot 2022-01-18 at 13 40 38

**Review by devnook — APPROVED** [review]
Please take a look at the caption alignment comment

**Pull request merged**